### PR TITLE
libgosu 0.10.7 (new formula) 

### PR DESCRIPTION
**Pull request by pmer**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

http://libgosu.org

Previous attempt from December 5th, 2014 at https://github.com/Homebrew/legacy-homebrew/pull/34700

Issue from previous attempt was that the upstream wasn't providing a `.dylib`. Now they are.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Install is failing: https://bot.brew.sh/job/Homebrew%20Core/12010/version=any_sierra/testReport/junit/brew-test-bot/sierra/install_libgosu/

Looks like a `depends_on "pkg-config" => :build` is needed.

**Comment by pmer** [comment]
@MikeMcQuaid Thanks!

File: Formula/libgosu.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class Libgosu < Formula
+  desc "2D game development library"
+  homepage "https://libgosu.org"
+  url "https://github.com/gosu/gosu/archive/v0.10.7.tar.gz"
+  sha256 "31aedcb570a36c344b9a3c12ea13596f994cb52a6fb8c15b8ed83058d9cabe27"
+
+  head do
+    url "https://github.com/gosu/gosu.git"
+  end
```
Comment (MikeMcQuaid) [inline Formula/libgosu.rb]: Just make this `head "https://github.com/gosu/gosu.git"`

File: Formula/libgosu.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "sdl2"
+
+  def install
+    mkdir "cmake/build"
+    cd "cmake/build" do
```
Comment (MikeMcQuaid) [inline Formula/libgosu.rb]: `mkdir "cmake/build" do`. Also, I'd favour just `build` unless this needs to be in the `cmake` directory (no worries if it does).

File: Formula/libgosu.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    Pathname("test.cpp").write <<-EOS.undent
```
Comment (MikeMcQuaid) [inline Formula/libgosu.rb]: `(testpath/"test.cpp").write <<-EOS.undent`

File: Formula/libgosu.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      };
+
+      int main()
+      {
+          MyWindow window;
+          window.show();
+      }
+    EOS
+
+    system ENV.cxx, "test.cpp", "-L#{lib}", "-lgosu", "-I#{include}", "-std=c++11"
```
Comment (MikeMcQuaid) [inline Formula/libgosu.rb]: Would be good to also run the outputted executable.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/libgosu.rb (hunk 4 above)
Comment (pmer) [inline Formula/libgosu.rb]: I've updated the test program to quit as soon as the window appears on-screen.

**Comment by pmer** [comment]
@MikeMcQuaid Should be ready for another look.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**